obs_screen_recognition_script.py

```diff
@@ -84,19 +84,26 @@ def frame_contains_one_or_more_matching_images(frame, mask, image_descriptors, f
                 return (True, len(good))
     return (False, len(good))
 
-def main():
-    with open("obs_screen_recognition_settings.json") as settings_file:
+@click.command()
+@click.option('--show-debug-window', is_flag=True)
+@click.argument('resource-dir', type=click.Path(exists=True,file_okay=False, dir_okay=True))
```

Comment:
This is really nice, thanks!

Reply:
we can easily drop the settings file in favour of full options set if you like, with proper defaults:
```
$ python obs_obfuscator.py --help
Usage: obs_obfuscator.py [OPTIONS] RESOURCE_DIR

Options:
  --show-debug-window
  --screen INTEGER        Index of the screen to capture
  --format [1080p|1440p]  Screen format
  --scene-on TEXT         OBS Scene to activate when triggered
  --scene-off TEXT        OBS Scene to activate when not triggered
  --features INTEGER      Number of features to detect (greater means better, but also slower)
  --matches INTEGER       Number of matches required for triggering (greater means more accurate, but slower)
  --help                  Show this message and exit.
```

thus you can launch the thing with something like

`$ python obs_obfuscator.py --format 1080p ./hll`

Reply:
I have this now in my `master` branch, which verifies if the resolution is supported just checking the presence of the required files and folders and uses no settings file:

```
$ python obs_obfuscator.py --help
Usage: obs_obfuscator.py [OPTIONS] RESOURCE_DIR

Options:
  --show-debug-window
  --screen <int>                  Index of the screen to capture  [default: 1]
  --format [720p|960p|1080p|1440p|2160p]
                                  Screen format  [required]
  --scene-on <text>               Scene to activate when triggered (use quotes
                                  if necessary)  [default: Live Gaming (Map
                                  Covered)]

  --scene-off <text>              Scene to activate when not triggered (use
                                  quotes if necessary)  [default: Live Gaming]

  --features <int>                Number of features to detect  [default: 500]
  --matches <int>                 Number of matches required for triggering
                                  [default: 20]

  --help                          Show this message and exit.

```

I'm going to use it as `python obs_obfuscator.py --format 1080p hll`

If you like this I can submit a PR 😉

Reply:
Personally I would prefer to have settings able to be stored in a config file, in order to minimise the amount of state that must be stored in exact command I invoke. CLI options are great, but I'd like to be able to set them in a config file too.

Did you also rename the script in your branch? Why `obs_obfuscator`? Edit: Ohh, I just figured out it's obfuscating the map in HLL, fair enough. (I'm hoping to keep this repo generic enough that it can be used to change OBS scenes in any game when specific content is shown).

Reply:
I perfectly understand your point with regards to the "minimise the amount of state", that's why I went for having a launch script (`hll_obfuscator.bat`) to launch by double-clicking on an icon:
![image](https://user-images.githubusercontent.com/199473/93828079-ee1f8f00-fc6a-11ea-8789-51632d4d8e17.png)

Batch script content is just a single line:
```
python obs_obfuscator.py --format 1080p hll
```

As I'm also very lazy, I made it possible to set each and every option/argument via an environment variable:
```
OBFUSCATOR_SCREEEN
OBFUSCATOR_FORMAT
OBFUSCATOR_SCENE_ON
OBFUSCATOR_SCENE_OFF
OBFUSCATOR_FEATURES
OBFUSCATOR_MATCHES
```

But I would perfectly understand if you want to stick with the settings file: I did all this just to play with the library, not because it was _**needed**_ in any way